developer/ruby/FindMissingSWIGTypes.rb

```diff
@@ -1,13 +1,91 @@
 require 'json'
+require 'optparse'
 
-# Change this to match your settings
-BUILD_DIRECTORY = '../../build'
+# define the input parameters
+options = Hash.new
+
+optparse = OptionParser.new do |opts|
+
+  opts.on( '-s', '--generated-sources-directory GENERATED_SOURCES_DIRECTORY',
+          String,
+          "Directory when the .cs generated files are, in build/csharp_wrapper/generated_sources") do |sourceDirectory|
+    options[:generatedSourcesDirectory] = sourceDirectory
+  end
+
+  opts.on( '-i', '--ignore-existing-errors', "Ignores the known errors") do |ignoreErrors|
+    options[:ignoreErrors] = ignoreErrors
+  end
+
+  opts.on( '-v', '--verbose', "Print errors that have been ignored") do |verbose|
+    options[:verbose] = verbose
+  end
+end
+
+# parse the input parameters
+optparse.parse!
+
+if options[:generatedSourcesDirectory]
+  GENERATED_SOURCE_DIRECTORY = options[:generatedSourcesDirectory]
+else
+  GENERATED_SOURCE_DIRECTORY = File.absolute_path(File.join(File.dirname(__FILE__), "../../build/csharp_wrapper/generated_sources"))
+  puts "Trying to default the generated-sources-directory to #{GENERATED_SOURCE_DIRECTORY}"
+end
+
+if !File.exist?(GENERATED_SOURCE_DIRECTORY)
+  puts "The generated-sources-directory '#{GENERATED_SOURCE_DIRECTORY}' does not exist. Pass it via `-s DIR`"
+  puts ""
+  puts optparse.help
+  exit 1
+end
+
+# This list was put together on 2021-02-23, based on 988dd3fcb92a7eea4a879e9543d4e2132c8ffe43
+known_errors = {
+  "OpenStudioUtilitiesCore" => [
+    "SWIGTYPE_p_openstudio__WorkspaceObject.cs",
+    "SWIGTYPE_p_DateTime.cs",
+    "SWIGTYPE_p_std__thread__id.cs",
+    "SWIGTYPE_p_openstudio__BCLComponent.cs"
+  ],
```

Comment:
I store a list of "known" errors. Eventually we should probably either fix all of them or just ignore them in SWIG interface files, but this'll do for now.